addons/binding/org.openhab.binding.synopanalyzer/src/main/java/org/openhab/binding/synopanalyzer/handler/SynopAnalyzerHandler.java

```diff
@@ -0,0 +1,227 @@
+                    case TEMPERATURE:
+                        state = new DecimalType(synop.getTemperature());
+                        break;
+                    case WIND_ANGLE:
+                        state = new DecimalType(synop.getWindDirection());
+                        break;
+                    case WIND_DIRECTION:
+                        int angle = synop.getWindDirection();
+                        String direction = getWindDirection(angle);
+                        state = new StringType(direction);
+                        break;
+                    case WIND_SPEED_MS:
+                        if (synop.getWindUnit().equalsIgnoreCase("m/s")) {
+                            state = new DecimalType(synop.getWindSpeed());
+                        } else {
+                            Double kmhSpeed = knotsToKmh(new Double(synop.getWindSpeed()));
+                            state = new DecimalType(kmhToMps(kmhSpeed));
+                        }
+                        break;
+                    case WIND_SPEED_KNOTS:
+                        if (synop.getWindUnit().equalsIgnoreCase("knots")) {
+                            state = new DecimalType(synop.getWindSpeed());
+                        } else {
+                            Double kmhSpeed = mpsToKmh(new Double(synop.getWindSpeed()));
+                            Double knotSpeed = kmhToKnots(kmhSpeed);
+                            state = new DecimalType(knotSpeed);
+                        }
+                        break;
+                    case WIND_SPEED_BEAUFORT:
+                        Double kmhSpeed;
+                        if (synop.getWindUnit().equalsIgnoreCase("m/s")) {
+                            kmhSpeed = mpsToKmh(new Double(synop.getWindSpeed()));
+                        } else {
+                            kmhSpeed = knotsToKmh(new Double(synop.getWindSpeed()));
+                        }
+                        Double beaufort = kmhToBeaufort(kmhSpeed);
+                        state = new DecimalType(beaufort);
+                        break;
+                    case TIME_UTC:
+                        Calendar observationTime = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+                        observationTime.set(Calendar.DAY_OF_MONTH, synop.getDay());
+                        observationTime.set(Calendar.HOUR_OF_DAY, synop.getHour());
+                        observationTime.set(Calendar.MINUTE, 0);
+                        observationTime.set(Calendar.SECOND, 0);
+                        state = new DateTimeType(observationTime);
+                        break;
+                    default:
+                        logger.debug("Unsupported channel Id '{}'", channelId);
+                        break;
+                }
+                updateState(channel.getUID(), state);
+            }
+        } catch (IOException e) {
+            logger.error("Error occured while receiving last Synop message", e);
```

Comment:
When can we get here? Only if there's a bug in your code? Then error with stacktrace is ok.
If this can happen upon network issues, set the Thing status to OFFLINE instead and remove the logging.